include/boost/geometry/algorithms/detail/envelope/linestring.hpp

```diff
@@ -108,7 +116,9 @@ struct envelope_linear_on_spheroid
             Box segment_mbr;
             dispatch::envelope
                 <
-                    typename std::iterator_traits<iterator_type>::value_type
+                    typename std::iterator_traits<iterator_type>::value_type,
+                    0,
+                    DimensionCount
```

Comment:
You are right. Either here I would need to concentrate on the first two dimensions, or keep things as they are and omit the dispatch call at the end.